t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -1122,6 +1122,34 @@ test_expect_success 'clean' '
 	test_sparse_match test_path_is_dir folder1
 '
 
+test_expect_success 'show (cached blobs/trees)' '
+	init_repos &&
+
+	test_all_match git show :a &&
+	test_all_match git show :deep/a &&
+	test_sparse_match git show :folder1/a &&
+
+	# Asking "git show" for directories in the index
+	# does not work as implemented. The error message is
+	# different for a full checkout and a sparse checkout
+	# when the directory is outside of the cone.
```

Comment:
Probably not necessary for now, but I'm guessing this should be fixed before submitting upstream? Most likely aligning the behavior of 'full-checkout', 'sparse-checkout', and 'sparse-index' either with an error or by "fixing" it in the non-sparse index cases.

Reply:
We definitely have time to think about it since the stuff currently sent upstream is taking a while to merge down. I'm personally torn between sending this as-is and asking "Should we align these behaviors?" versus doing the change in the same series and asking "Should we include patch X?"

I will think on this.